src/Compiler/CompilerPasses.cpp

```diff
@@ -45,8 +45,8 @@ namespace onnx_mlir {
 void configurePasses() {
   // Set global vector machine support.
   VectorMachineSupport::setGlobalVectorMachineSupport(march, mcpu, "");
-  configureConstPropONNXToONNXPass(
-      onnxConstPropExpansionBound, onnxConstPropDisablePatterns);
+  configureConstPropONNXToONNXPass(onnxConstPropExpansionBound,
+      onnxConstPropDisablePatterns, OptimizationLevel, enableConstantProp);
```

Comment:
thanks, this solution satisfies my concerns about dependencies

fwiw, configureConstPropONNXToONNXPass could potentially omit the optLevel parameter if you passed a single value `(OptimizationLevel >= 3 || enableConstProp)` for constantPropIsEnabled here - that way ConstPropONNXToONNXPass wouldn't need to know the details of the policy for enabling/disabling it

@AlexandreEichenberger please go ahead and approve if your concerns are addressed

Reply:
I missed @sorenlassen suggestion, and in fact I like it. Can we remove `OptimizationLevel` from the `ConstPropONNXToONNXPassConfiguration` object, simply call  here
```
 configureConstPropONNXToONNXPass(onnxConstPropExpansionBound,
       onnxConstPropDisablePatterns, OptimizationLevel>=3 ||  enableConstantProp);
```

as really, the const propagation pass does not need to be aware of if/when -O flags matter. That way, all this logic is neatly hidden in one spot.

Last chance, promised.

Reply:
@AlexandreEichenberger Done